cosmos-sdk-rs/src/base.rs

```diff
@@ -71,15 +75,26 @@ impl FromStr for AccountId {
     type Err = eyre::Report;
 
     fn from_str(s: &str) -> Result<Self> {
-        let (hrp, bytes) = bech32::decode(s)?;
+        let (hrp, bytes) = bech32::decode(s).map_err(|e| Error::AccountId {
+            id: s.to_owned(),
+            detail: format!("failed to decode bech32: {}", e.to_string()),
+        })?;
 
         if bytes.len() == tendermint::account::LENGTH {
             Ok(Self {
                 bech32: s.to_owned(),
                 hrp_length: hrp.len(),
             })
         } else {
-            Err(Error::AccountId { id: s.to_owned() }.into())
+            Err(Error::AccountId {
+                id: s.to_owned(),
+                detail: format!(
+                    "account ID should be at least {} bytes long, but was {} bytes long",
+                    tendermint::account::LENGTH,
+                    bytes.len()
+                ),
+            }
+            .into())
```

Comment:
FYI, since we're using `eyre`, it already provides the functionality to attach arbitrary messages like this to errors without requiring each variant carry along a `String` field:

```suggestion
use eyre::WrapErr;

Err(Error::Address).wrap_err_with(|| {
    format!(
        "account ID should be at least {} bytes long, but was {} bytes long",
        tendermint::account::LENGTH,
        bytes.len()
    ),
})
```

Reply:
Done!